From 2fcec283c976ac7979414a5df1eaf67142d5c74f Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 29 Nov 2017 13:27:23 +0200
Subject: [PATCH 2938/5242] MLK-17032-2: drm: imx: dcss: fix runtime
 suspend/resume

commit  7b5454ba01e8ad0f0b09bc5021939a71c62246ee from
https://source.codeaurora.org/external/imx/linux-imx.git

If the DCSS core is runtime suspended, but the display-subsystem is not,
we need to resume the DCSS core before setting up DTG and SUBSAM
modules.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-crtc.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-crtc.c b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
index eaf7296..184eb29 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-crtc.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
@@ -128,9 +128,14 @@ static void dcss_crtc_mode_set_nofb(struct drm_crtc *crtc)
 
 	drm_display_mode_to_videomode(mode, &vm);
 
+	pm_runtime_get_sync(dcss_crtc->dev->parent);
+
 	dcss_dtg_sync_set(dcss, &vm);
 	dcss_ss_sync_set(dcss, &vm, mode->flags & DRM_MODE_FLAG_PHSYNC,
 			 mode->flags & DRM_MODE_FLAG_PVSYNC);
+
+	pm_runtime_mark_last_busy(dcss_crtc->dev->parent);
+	pm_runtime_put_autosuspend(dcss_crtc->dev->parent);
 }
 
 static int dcss_crtc_atomic_check(struct drm_crtc *crtc,
-- 
1.7.9.5

